contracts/facets/EmbalmerFacet.sol

```diff
@@ -0,0 +1,130 @@
+        address arweaveArchaeologist,
+        address recipient,
+        uint256 resurrectionTime,
+        bytes32 identifier,
+        IERC20 sarcoToken,
+        bool canBeTransfered
+    ) external returns (uint256) {
+        // Confirm that this exact sarcophagus does not already exist
+        require(
+            !s.sarcophaguses[identifier].exists,
+            "sarcophagus already exists"
+        );
+
+        // Confirm that the ressurection time is in the future
+        require(
+            resurrectionTime > block.timestamp,
+            "resurrection time must be in the future"
+        );
+
+        // Initialize a list of archaeologist addresses
+        address[] memory archaeologistAddresses = new address[](
+            archaeologists.length
+        );
+
+        for (uint256 i = 0; i < archaeologists.length; i++) {
+            // Calculate the amount of cursed bond the archaeologists needs to lock up
+            uint256 cursedBondAmount = LibBonds.calculateCursedBond(
+                archaeologists[i].diggingFee,
+                archaeologists[i].bounty
+            );
+
+            // Confirm that the archaeologist has enough free bond.
+            // This error could mean that the archaeologist has either run out
+            // of free bond or has never even interacted with sarcophagus
+            require(
+                s.freeBonds[archaeologists[i].archAddress] >= cursedBondAmount,
+                "archaeologist does not have enough free bond"
+            );
+
+            // Lock up the archaeologist's bond by the cursed bond amount
+            LibBonds.lockUpBond(
+                archaeologists[i].archAddress,
+                cursedBondAmount
+            );
+
+            // Add to the neccessary data structures
+            s.archaeologistSarcophaguses[archaeologists[i].archAddress].push(
+                identifier
+            );
+            s.sarcophagusArchaeologists[identifier][
+                archaeologists[i].archAddress
+            ] = archaeologists[i];
+
+            // Add the archaeologist address to the list of addresses to be
+            // passed in to the sarcophagus object
+            archaeologistAddresses[i] = archaeologists[i].archAddress;
+        }
+
+        s.sarcophaguses[identifier] = LibTypes.Sarcophagus({
+            name: name,
+            exists: true,
+            canBeTransferred: canBeTransfered,
+            resurrectionTime: resurrectionTime,
+            arweaveTxId: "",
+            embalmer: msg.sender,
+            recipientAddress: recipient,
+            arweaveArchaeologist: arweaveArchaeologist,
+            archaeologists: archaeologistAddresses
+        });
+
+        // Add the identifier to necessary data structures
+        s.sarcophagusIdentifiers.push(identifier);
+        s.embalmerSarcophaguses[msg.sender].push(identifier);
+        s.recipientSarcophaguses[recipient].push(identifier);
```

Comment:
Did we decide to leave these storage mappings off the contract (`archaeologist/embalmer/recipient sarcphaguses`), and instead use events filtering for the front-end to display these?

Reply:
It didn't seem like we came to a definite conclusion on that discussion so I left it in here. I agree though that we should try to get this information from the events.

Reply:
We can get this info via events, so I think these can be removed now, or can put in a TODO to remove them later.